File: Java_Programming/Practise_Programs/PP_029.java
//////////////////////////////////////////////////////////////////
//
//  File Name         : PP_029
//
//  Description       : This program accepts a number from the user
//                      and counts the occurrence of digit 7 in
//                      the given number.
//
//  Author            : Shubham Somanath Gadhe
//  Date              : 10/07/2026
//
//  Time Complexity   : O(n)
//  Space Complexity  : O(1)
//
//////////////////////////////////////////////////////////////////

import java.util.Scanner;

class DigitX
{
    //////////////////////////////////////////////////////////////////
    //
    //  Method Name      : CountDigits()
    //
    //  Description      : It is used to count the occurrence of
    //                     digit 7 in the given number by extracting
    //                     each digit using modulus and division.
    //
    //  Author           : Shubham Somanath Gadhe
    //  Date             : 10/07/2026
    //
    //////////////////////////////////////////////////////////////////

    public int CountDigits(int iNo)
    {
        int iDigit = 0;
        int iCount = 0;

        while(iNo != 0)
        {
            iDigit = iNo % 10;

            if(iNo == 7)
            {
                iCount++;
            }

            iNo = iNo / 10;
        }

        return iCount;
    }
}

class PP_029
{
    public static void main(String A[])
    {
        Scanner sobj = new Scanner(System.in);
        DigitX dobj = new DigitX();

        int iValue = 0;
        int iRet = 0;

        System.out.println("Enter Number : ");
        iValue = sobj.nextInt();

        iRet = dobj.CountDigits(iValue);

        System.out.println("Number of Digits are : " + iRet);
    }
}

//////////////////////////////////////////////////////////////////
//
//  Sample Input  :
//      Enter Number :
//      127737
//
//  Sample Output :
//      Number of Digits are : 1
//
//////////////////////////////////////////////////////////////////

File: C++_Programming/Practise_Programs/PP_029.cpp
//////////////////////////////////////////////////////////////////
//
//  File Name         : PP_029
//
//  Description       : Accepts two numbers from the user and
//                      calculates and displays their sum.
//
//  Author            : Shubham Somanath Gadhe
//  Date              : 17/09/2026
//
//  Time Complexity   : O(1)
//  Space Complexity  : O(1)
//
//////////////////////////////////////////////////////////////////

#include<iostream>
using namespace std;

int main() 
{
    int num1, num2, sum;

    cout << "Enter the first number: ";
    cin >> num1; 

    cout << "Enter the second number: ";
    cin >> num2; 

    sum = num1 + num2; 

    cout << "The sum of the two numbers is: " << sum << endl;

    return 0;
}

//////////////////////////////////////////////////////////////////
//
//  Sample Input :
//      Enter the first number: 10
//      Enter the second number: 20
//
//  Sample Output :
//      The sum of the two numbers is: 30
//
//////////////////////////////////////////////////////////////////